test/ERC6538Registry.t.sol

```diff
@@ -47,24 +46,296 @@ contract RegisterKeys is ERC6538RegistryTest {
+      SigUtils.RegistrantInfo memory registrantInfo =
+        SigUtils.RegistrantInfo(alice, schemeId, stealthMetaAddress, nonce);
+      bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+      (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+      bytes memory signature = abi.encodePacked(r, s, v);
+
+      vm.expectEmit(true, true, true, true);
+      emit StealthMetaAddressSet(alice, schemeId, stealthMetaAddress);
+      registry.registerKeysOnBehalf(alice, schemeId, signature, stealthMetaAddress);
+    }
+  }
+
+  function testFuzz_RevertIf_ERC712SignatureIsNotValid(
+    string memory name,
+    address bob,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress
+  ) external {
+    (address alice, uint256 alicePk) = makeAddrAndKey(name);
+    SigUtils.RegistrantInfo memory registrantInfo =
+      SigUtils.RegistrantInfo(alice, schemeId, stealthMetaAddress, 0 /* nonce */ );
+    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    vm.expectRevert();
+    registry.registerKeysOnBehalf(bob, schemeId, signature, stealthMetaAddress);
+  }
+
+  function testFuzz_RevertIf_ERC1271SignatureIsNotValid(
+    string memory name,
+    address bob,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress
+  ) external {
+    ( /* address alice */ , uint256 alicePk) = makeAddrAndKey(name);
+
+    vm.prank(bob);
+    ERC1271MockContract erc1271MockContract = new ERC1271MockContract();
+    address registrant = address(erc1271MockContract);
+
+    SigUtils.RegistrantInfo memory registrantInfo =
+      SigUtils.RegistrantInfo(registrant, schemeId, stealthMetaAddress, 0 /* nonce */ );
+    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+
+    vm.expectRevert("Invalid signature");
+    registry.registerKeysOnBehalf(registrant, schemeId, signature, stealthMetaAddress);
+  }
+
+  function testFuzz_RevertIf_WrongNonce(
+    string memory name,
+    uint256 schemeId,
+    bytes memory stealthMetaAddress,
+    uint256 nonce
+  ) external {
+    vm.assume(nonce != 0);
+    (address alice, uint256 alicePk) = makeAddrAndKey(name);
+    SigUtils.RegistrantInfo memory registrantInfo =
+      SigUtils.RegistrantInfo(alice, schemeId, stealthMetaAddress, nonce);
+    bytes32 hash = sigUtils.getTypedDataHash(registrantInfo);
+    (uint8 v, bytes32 r, bytes32 s) = vm.sign(alicePk, hash);
+    bytes memory signature = abi.encodePacked(r, s, v);
+    vm.expectRevert();
+    registry.registerKeysOnBehalf(alice, schemeId, signature, stealthMetaAddress);
+  }
+
+  function test_RevertIf_NoSignatureIsProvided() external {
+    vm.expectRevert();
     registry.registerKeysOnBehalf(address(0), 0, "", "");
   }
+
+  function testFuzz_IncrementNonceCorrectly(address registrant) external {
```

Comment:
This test should be in it's own contract named `IncrementNonce` and should have a more descriptive name that plays well with `scopelint spec` generation

Reply:
Update in [a5b3e4d](https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/a5b3e4db42c7230a4ae1eab0e8bca6c723c0ea1d)